client/lib/purchases/index.ts

```diff
@@ -253,7 +253,12 @@ export function getDisplayName( purchase: Purchase ): TranslateResult {
 	const { productName, productSlug, purchaseRenewalQuantity } = purchase;
 	const jetpackProductsDisplayNames = getJetpackProductsDisplayNames( 'full' );
 
-	if ( isJetpackAISlug( purchase.productSlug ) && purchase.purchaseRenewalQuantity ) {
+	if (
+		isJetpackAISlug( purchase.productSlug ) &&
+		purchase.purchaseRenewalQuantity &&
+		purchase.priceTierList &&
+		purchase.priceTierList.length > 0
```

Comment:
Possible improvement that I think should work the same
```suggestion
		purchase.priceTierList?.length > 0
```

Reply:
The thing is typescript doesn't allow for direct comparison of `undefined` and numbers. What could work though is the below 🤔 What do you think?
```suggestion
		! purchase.priceTierList?.length
```